### ISSUE TITLE
A code of conduct for the Agda community?

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#3608]
I believe the Agda community is in general really nice and friendly to newcomers. (Please let me know if you're a newcomer and you found that's not the case!) Since I hope the community will grow in the future and I would like it to stay a welcoming place, I think it would be a good idea to have a code of conduct for interactions here on Github, on the mailing list, and at the Agda meetings.

Github has some [advice on adopting a code of conduct](https://opensource.guide/code-of-conduct/) and recommends the [contributor covenant](https://www.contributor-covenant.org/) as a baseline. One thing we should think about is to have one or more persons who can be contacted in case of a violation.

As I have no previous experience with setting up such a thing, any help and suggestions would be very much appreciated.

**Comment by xekoukou** [comment]
In my opinion, the main problem with codes of conducts is that the entity that enforces them is different from the community. This leads to the question of what can be considered as the agda community, what are the goals of the community etc.

  **In order to avoid those difficult questions, I think that we need to define an objective method of who is part of the governing body. One such objective method is a function that takes as arguments the number of commits , the number of lines added etc. to the project. This can measure the commitment of a person to the project.**
  This measure does not take into account the people that use the project, which are very different than the the developers. The developers, are mostly type theorists, the users are mathematicians and probably some programmers. So they have different motivations.

  But for the simple case of a code of conduct, this simple **objective** and **transparent** function that defines the governing body is enough.

**Comment by andreasabel** [comment]
Is there any pretext for necessitating a code of conduct for our github community?  
The absence of a dedicated code does not mean _anything goes_.  Common sense and high moral standards apply by default.  We can take measures against someone violating good behavioral standards even without a written code of conduct.
Btw, this gives me an opportunity to remind all developers of this code of conduct:
https://github.com/andreasabel/haskell-style-guide/

**Comment by jespercockx** [comment]
> In order to avoid those difficult questions, I think that we need to define an objective method of who is part of the governing body. One such objective method is a function that takes as arguments the number of commits , the number of lines added etc. to the project. This can measure the commitment of a person to the project.

I agree in principle, however I think regular contributors who do *not* want to deal with community management shouldn't be forced to. So I would take both commitment to the project *and* willingness as necessary qualifications.

> Is there any pretext for necessitating a code of conduct for our github community? 

No, as I said I believe the Agda community is a very healthy one. In an ideal situation, the code of conduct would just sit in the repo and be read by newcomers, but we would never have to enforce it. In case anything happens having a code of conduct means we don't have to come up with rules on the spot.

While we don't have a problem with bad community members, we **do** have a huge diversity problem. This is very clear from the Agda meetings. Having a code of conduct would hopefully be a first step to make people feel safe to join the community, ask questions, and even start contributing. For me this is the main reason.

Here is a quote by @ejgallego on the Coq github (https://github.com/coq/coq/issues/6477) that I strongly agree with:

> The fact that you and me don't have the need for it doesn't imply that it is not extremely important to atract a different profile of collaborator. We both belong to an extremely privileged class in the sense that we are relatively sure of what we do, and we go eating trolls anyways when our postdoc salary means that we don't meet ends at the end of the month. At least for myself, every time a BOFH-wannabe reminds me of my admitedly large ignorance about make or bash-quoting I smile and think "Emilio, you have done the right choices in life!"
> 
> I'd warn about dismissing the voices asking for more civil (even if reglamented) discourse as "hysterical". Again, a word that is perfectly fine for us won't have the same effect for people not so integrated in the community.
> 
> Let me remind that last time I asked I got a total consensus on that our technology and our community were seen as "extremely intimidating", and IMVHO we absolutely ought to change that.
> 
> An intimidating community acts as a kind of death filter, Egyptian-style. Yes, only bold and proactive people will feel able to join, but in the end them alone wont suffice to keep it alive.

**Comment by MatthewDaggitt** [comment]
Just a heads-up we did have an incident in https://github.com/agda/agda-stdlib/issues/595 where a newcomer mentioned that they felt unwelcome in the community due to the title of the issue in question. Hopefully it was resolved to the user's satisfaction (although we didn't get direct feedback).

**Comment by Saizan** [comment]
The contributor covenant code of conduct seems pretty good, it covers a lot of bases while being short and not overly specific.

https://www.contributor-covenant.org/version/1/4/code-of-conduct.txt


As for enforcement it could be enough to have 2-3 people assigned, as in the example here:

https://opensource.guide/code-of-conduct/#deciding-how-youll-enforce-your-code-of-conduct

**Comment by pnlph** [comment]
I am grateful that i found this thread. I like also the contributor covenant that Github recommends. If you like it, it may be a good idea to add it to the badges in the readme for better visibility.

[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v1.4%20adopted-ff69b4.svg)](https://www.contributor-covenant.org/version/1/4/code-of-conduct)

Apart from the code of conduct, i miss also a contributor guide that let members with different backgrounds know what they can do to contribute and to which extent. I mean, practically give some choices or advice on how they could commit their first contribution "today", which topics are open to which level of expertise, which book you should read before touching the code in that folder...

In my opinion, Agda is a potentially accesible language that allows the reasoning process to be clear and expressive but joining the community can be quite intimidating for almost everybody. From outside, it seems like a newbie here have at least a type-theory-related PhD or is on his/her way to achieve it. This is a contrast with the formalization possibilities that Agda offers regarding the reason ability that we all enjoy and use as humans on a daily basis.

I believe that, despite its strong scholar character, it could be used eg. in school. Also different user's perspectives would enrich and widespread its use.

I got into Agda by reading @astump book on functional programming verification and [IAL](https://github.com/cedille/ial). It is exactly the kind of language i was looking for, where every concept can easily be traced back to its foundational primitives. I still did not contribute, but i can say i did not feel not welcomed.

Just my 2 cents.
